docs/rfcs/001-updateable-bundled-packages.md

```diff
@@ -0,0 +1,100 @@
+> - Explaining any changes to existing workflows.
+
+Any bundled Atom package can opt in to new updates released via `apm` by adding `"coreUpdateable": true` to its `package.json` file.  This causes `apm` to consider it as part of the list of packages it checks for updates.  If a community (non-bundled) package sets this field to `true` or `false` it will be ignored as it's only relevant to bundled packages.
+
+Atom shows update notifications for updateable bundled packages whenever they are available so long as those updates support the engine version of the current Atom build.  Bundled package updates can also be found and installed in the Settings view's *Updates* tab.
+
+The `dalek` package is aware of the new "updateable" metadata and excludes updated bundled packages from its deprecation warnings.
+
+### User Experience Examples
+
+1. The user downloads and installs Atom 1.28.0 which includes GitHub package version 0.15.0.  Two weeks later, GitHub package 0.16.0 is released with a few new features.  The user is prompted to update to the new version and gets the new features even though Atom 1.29.0 hasn't been released yet.
+
+2. The user downloads and installs Atom 1.28.0, including GitHub package 0.15.0, which was released two weeks prior.  Since that release the GitHub package has been updated to version 0.15.1 on `apm`.  When the user starts Atom for the first time they are prompted to update the GitHub package.
+
+3. In the future, a user has an old install of Atom 1.28.0 and waits a long time between installing Atom updates.  The GitHub package releases version 0.25.0 but the user is not prompted to install it because the GitHub package has set `engines` in `package.json` to restrict to Atom 1.32.0 and above.
+
+### Rules for Updateable Bundled Packages
+
+Any package that opts into this behavior must adhere to these rules:
+
+1. **Each release must ensure that its `engines` field in `package.json` reflects the necessary Atom version for the Atom, Electron, and Node.js APIs used in the package**.  This field defines the range of Atom versions in which the package is expected to work.  The field should always be set to the lowest possible Atom version that the package supports.
+
+2. **Any new update to a bundled package *must* support current Stable *and* Beta releases**.  This enables the user to upgrade the package and continue to use it in side-by-side Stable and Beta installs on their machine.  If a package wants to use API features of a newer version of Atom while still supporting older Atom versions, it must do so in a way that is aware of the user's version and adjust itself accordingly.
+
+3. **Atom's `package.json` *must* stay up to date with the latest supported version of the package** in the `master` and Beta release branches.  This ensures that the user always gets the latest version of the package in a new release and also benefits from its inclusion in Atom's snapshot.
```

Comment:
@daviwil: I think I understand the motivation for this rule, but if a human needs to manually bump the package version on atom's master and beta branches each time a new version of the package is released, I'm skeptical that this rule will be reliably followed. I think we'd need some automation to either perform this step for us, or alert us when this rule is violated so that we can remedy the situation.

What do you think?

Reply:
Yep, it'd be nice if we had something like [Greenkeeper](https://greenkeeper.io/) for Atom packages.  Actually, I think it wouldn't be hard to have a special build automation for that which only runs a check against known updatable bundled packages (i.e. GitHub and possibly Teletype) and bumps Atom's package.json dependencies when a newer compatible version is available.

I'll add a blurb about that to the RFC.